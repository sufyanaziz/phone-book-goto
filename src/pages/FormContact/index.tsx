import React, { Suspense } from "react";
import Container from "@common/components/Container";
import HeaderTitle from "@common/components/HeaderTitle";

const Form = React.lazy(() => import("./components/Form"));

const FormContact = () => {
  return (
    <Container
      css={{
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        flexDirection: "column",
      }}
    >
      <div
        css={{
          width: "85vw",
          border: "1px solid black",
          padding: "16px",
          "@media (min-width: 720px)": {
            width: "50vw",
          },
        }}
      >
        <HeaderTitle
          title="Form Contact"
          css={{ display: "flex", justifyContent: "center" }}
        />
        <Suspense fallback={<p>Loading...</p>}>
          <Form />
        </Suspense>
      </div>
    </Container>
  );
};

export default FormContact;
