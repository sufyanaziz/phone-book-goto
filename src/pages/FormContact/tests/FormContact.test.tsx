import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { ApolloClient, ApolloProvider, InMemoryCache } from "@apollo/client";
import FormContact from "@pages/FormContact";

const client = new ApolloClient({
  uri: "",
  cache: new InMemoryCache(),
});

describe("Component FormContact", () => {
  it("should render FormContact component correctly", () => {
    const component = renderer.create(
      <ApolloProvider client={client}>
        <Router>
          <FormContact />
        </Router>
      </ApolloProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
