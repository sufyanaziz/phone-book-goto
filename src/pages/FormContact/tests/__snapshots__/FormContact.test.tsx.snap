// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component FormContact should render FormContact component correctly 1`] = `
<div
  css={[Function]}
>
  <div
    css={
      {
        "@media (min-width: 720px)": {
          "width": "50vw",
        },
        "border": "1px solid black",
        "padding": "16px",
        "width": "85vw",
      }
    }
  >
    <p>
      Loading...
    </p>
  </div>
</div>
`;
